client/src/util/auth.jsx

```diff
@@ -30,243 +42,228 @@ function useAuthProvider() {
-      throw new Error(
-        "Thanks for signing up! Please check your email to complete the process.",
-      );
-    }
+  //!whats the point of this
+  const handleAuth = async (user) => {
 
     // Update user in state
     setUser(user);
     return user;
   };
 
-  const signup = (email, password) => {
-    return supabase.auth
-      .signUp({ email, password })
-      .then(handleError)
-      .then(handleAuth);
-  };
+  const signup = async (email, password) => {
+    try {
+      const userCredential = await createUserWithEmailAndPassword(
+        auth,
+        email,
+        password,
+      );
 
-  const signin = (email, password) => {
-    return supabase.auth
-      .signInWithPassword({ email, password })
-      .then(handleError)
-      .then(handleAuth);
+      await handleAuth(userCredential.user);
+    } catch (error) {
+      handleError(error);
+    }
+    // return supabase.auth
+    //   .signUp({ email, password })
+    //   .then(handleError)
+    //   .then(handleAuth);
   };
 
-  const signinWithProvider = (name) => {
-    return (
-      supabase.auth
-        .signInWithOAuth({
-          provider: name,
-          options: {
-            redirectTo: `${window.location.origin}/dashboard`,
-          },
-        })
-        .then(handleError)
-        // Because `signInWithOAuth` resolves immediately we need to add this so that
-        // it never resolves (component will display loading indicator indefinitely).
-        // Once social signin is completed the page will redirect to value of `redirectTo`.
-        .then(() => {
-          return new Promise(() => null);
-        })
-    );
+  const signin = async (email, password) => {
+    try {
+      const userCredential = await signInWithEmailAndPassword(
+        auth,
+        email,
+        password,
+      );
+
+      await handleAuth(userCredential.user);
+    } catch (error) {
+      handleError(error);
+    }
+    // return supabase.auth
+    //   .signInWithPassword({ email, password })
+    //   .then(handleError)
+    //   .then(handleAuth);
   };
 
-  const signinWithMagicLink = (email) => {
-    return supabase.auth
-      .signInWithOtp({
-        email,
-        options: {
-          emailRedirectTo: `${window.location.origin}/dashboard`,
-        },
-      })
-      .then(handleError);
+  const signinWithProvider = async (name) => {
+
+    let provider;
+
+    switch (name) {
+      case 'google':
+        provider = new GoogleAuthProvider();
+        break;
+      case 'github':
+        provider = new GithubAuthProvider();
+        break;
+      default:
+        throw new Error(`Provider ${name} is not supported`);
+    }
+
+    try {
+      await signInWithPopup(auth, provider);
+
+      window.location.href = `${window.location.origin}/dashboard`;
+    } catch (error) {
+      console.log('Error signing in with provider', error.message);
+      handleError(error);
+    }
+
+    return new Promise(() => null);
```

Comment:
Why you are returning a Promise that resolves to null here? If this function returned null or threw an error, it would be more concise

Reply:
legacy code. Removed